src/pipes/input-characters-validation.test.ts

```diff
@@ -0,0 +1,171 @@
+    });
+
+    it('no error for simple Body values dto has unsupported characters in object properties', () => {
+      const functionToTest = () => {
+        pipe.transform(bodyDto, bodyMeta);
+      };
+
+      expect(functionToTest).not.toThrow();
+    });
+
+    it('throws an error if Body param dto has unsupported characters in object properties', () => {
+      const issueInLevel1 = { ...bodyDto, field1: bodyTextWithUnsupportedCharacters };
+      const functionToTest = () => {
+        pipe.transform(issueInLevel1, bodyMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+    });
+
+    it('throws an error if Body param dto has unsupported characters in object child properties', () => {
+      const issueInLevel2 = { ...bodyDto, field2: { field2_2: bodyTextWithUnsupportedCharacters } };
+      const functionToTest = () => {
+        pipe.transform(issueInLevel2, bodyMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+    });
+
+    it('throws an error if Body param dto has unsupported characters in object grandchild properties', () => {
+      const issueInLevel3 = { ...bodyDto, field3: { field3_3: { field3_3_2: bodyTextWithUnsupportedCharacters } } };
+      const functionToTest = () => {
+        pipe.transform(issueInLevel3, bodyMeta);
+      };
+
+      expect(functionToTest).toThrow(BadRequestException);
+    });
+
+    describe('Validate that known ASCII characters pass', () => {
+      for (let i = 32; i <= 126; i++) {
+        it('no error for ascii ' + i, () => {
+          const bodyToTest = { ...bodyDto, field1: String.fromCharCode(i) };
+          const functionToTest = () => {
+            pipe.transform(bodyToTest, bodyMeta);
+          };
+
+          expect(functionToTest).not.toThrow();
+        });
+      }
+      // ACBS doesn't support characters 181 µ and 255 ÿ
+      for (let i = 160; i <= 254; i++) {
+        if (i === 181) {
+          continue;
+        }
+
+        it('no error for ascii ' + i, () => {
+          const bodyToTest = { ...bodyDto, field1: String.fromCharCode(i) };
+          const functionToTest = () => {
+            pipe.transform(bodyToTest, bodyMeta);
+          };
+
+          expect(functionToTest).not.toThrow();
+        });
+      }
+
+      it('throws an error if Body param dto has ASCII character 181 in object properties', () => {
+        const bodyToTest = { ...bodyDto, field1: String.fromCharCode(181) };
+        const functionToTest = () => {
+          pipe.transform(bodyToTest, bodyMeta);
+        };
+
+        expect(functionToTest).toThrow(BadRequestException);
+      });
+
+      it('throws an error if Body param dto has ASCII character 255 in object properties', () => {
+        const bodyToTest = { ...bodyDto, field1: String.fromCharCode(255) };
+        const functionToTest = () => {
+          pipe.transform(bodyToTest, bodyMeta);
+        };
+
+        expect(functionToTest).toThrow(BadRequestException);
```

Comment:
Should we be testing the error message as well?

Reply:
The error message is always `Bad request`, would it be better to throw `throw new BadRequestException(errorMessageGenerator(key, invalidCharacters));` ?